Utilities/bootstrap

```diff
@@ -547,27 +547,40 @@ def get_swiftpm_flags(args):
             "-Xlinker", "-rpath",
             "-Xlinker", "@executable_path/../../../../../SharedFrameworks",
 
-            # For TensorFlow.
-            "-Xlinker", "-rpath",
-            "-Xlinker", "@executable_path/../lib/swift/macosx",
-
             # For LLBuild in CLT.
             "-Xlinker", "-rpath",
             "-Xlinker", "@executable_path/../lib/swift/pm/llbuild",
         ])
 
-    # Add a relative rpath to find core Swift libraries on non-Darwin platforms,
-    # where they aren't present as part of the base OS.
-    if platform.system() != 'Darwin':
-        platform_path = None
-        for path in args.target_info["paths"]["runtimeLibraryPaths"]:
-            platform_path = re.search(r'(lib/swift/[^/]+)$', path)
-            if platform_path:
-                build_flags.extend(["-Xlinker", "-rpath=$ORIGIN/../" + platform_path.group(1)])
-                break
-
-        if not platform_path:
-            error("the command `%s -print-target-info` didn't return a valid runtime library path" % args.swiftc_path)
+    # Add a relative rpath to find Swift libraries in toolchains.
+    # On non-Darwin platforms, a relative rpath is necessary because Swift
+    # libraries are not part of the OS.
+    # On Darwin platforms, a relative rpath is necessary for experimental
+    # toolchains that include libraries not part of the OS (e.g. PythonKit or
+    # TensorFlow).
+    if platform.system() == "Darwin":
+        swift_library_rpath_prefix = "@executable_path/../"
+    else:
```

Comment:
Please make this `elif platform == 'Linux' || platform.matches('BSD'):`.  `$ORIGIN` is an ELFish construct, it does not apply to Windows or AIX.

Reply:
Done in https://github.com/apple/swift-package-manager/pull/2553/commits/fd3f9823deb693f76ec22a0ca823be48cc22b20d.

Reply:
`platform.matches('BSD')` doesn't appear to work:
```
11:45:00 Traceback (most recent call last):
11:45:00   File "/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/swiftpm/Utilities/bootstrap", line 593, in <module>
11:45:00     main()
11:45:00   File "/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/swiftpm/Utilities/bootstrap", line 62, in main
11:45:00     args.func(args)
11:45:00   File "/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/swiftpm/Utilities/bootstrap", line 265, in build
11:45:00     build_swiftpm_with_swiftpm(args)
11:45:00   File "/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/swiftpm/Utilities/bootstrap", line 457, in build_swiftpm_with_swiftpm
11:45:00     "--disable-sandbox",
11:45:00   File "/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/swiftpm/Utilities/bootstrap", line 476, in call_swiftpm
11:45:00     full_cmd = get_swiftpm_env_cmd(args) + cmd + get_swiftpm_flags(args)
11:45:00   File "/home/buildnode/jenkins/workspace/swift-package-manager-Linux-smoke-test/swiftpm/Utilities/bootstrap", line 563, in get_swiftpm_flags
11:45:00     elif platform == 'Linux' or platform.matches('BSD'):
11:45:00 AttributeError: 'module' object has no attribute 'matches'
```

Reading the documentation for [`platform`](https://docs.python.org/2/library/platform.html), I don't see mention of `BSD` anywhere. So I simplified the condition to `elif platform == 'Linux':`.